modules/httpboot/httpboot_api.rb

```diff
@@ -0,0 +1,10 @@
+class Proxy::HttpbootApi < Sinatra::Base
+  helpers ::Proxy::Helpers
+
+  get "/*" do
+    file = Pathname.new(params[:splat].first).cleanpath
```

Comment:
Is this safe to prevent ../../../etc/shadow ?

Reply:
Sinatra filters it out, among other tricky ways to do path traversal.

Reply:
Yep I tested this, but I was wondering if we should allow following symlinks outside of TFTP root directory.

I can prevent from both possible security issues by comparing absolute file path with root directory - if it's not the same then we have an error. You know what, let me do that, just for case. I don't want my name on a CVE like this :-)

Reply:
I am doing this, we can't be sure if e.g. older version of Sinatra does this.